src/main/java/org/zaproxy/zap/extension/hud/launch/ExtensionHUDlaunch.java

```diff
@@ -0,0 +1,179 @@
+        Class<?> extSeleniumClass;
+        try {
+            extSeleniumClass = Class.forName(SELENIUM_EXTENSION_CLASS_NAME);
+            dependencies.add((Class<? extends Extension>) extSeleniumClass);
+            DEPENDENCIES = Collections.unmodifiableList(dependencies);
+        } catch (ClassNotFoundException e) {
+            DEPENDENCIES = null;
+        }
+    }
+
+    public ExtensionHUDlaunch() {
+        super(NAME);
+    }
+
+    @Override
+    public boolean supportsDb(String type) {
+        return true;
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+        extensionHook.addCommandLine(getCommandLineArguments());
+    }
+
+    @Override
+    public boolean canUnload() {
+        return true;
+    }
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
+
+    private Extension getExtSelenium() {
+        return Control.getSingleton()
+                .getExtensionLoader()
+                .getExtensionByClassName(SELENIUM_EXTENSION_CLASS_NAME);
+    }
+
+    private ExtensionHUD getExtHUD() {
+        return Control.getSingleton().getExtensionLoader().getExtension(ExtensionHUD.class);
+    }
+
+    private Object callMethodByReflection(Object obj, String methodName, Object... args)
+            throws IllegalAccessException, IllegalArgumentException, InvocationTargetException,
+                    NoSuchMethodException, SecurityException {
+        @SuppressWarnings("rawtypes")
+        Class[] argClasses = new Class[args.length];
+        int i = 0;
+        for (Object arg : args) {
+            argClasses[i] = arg.getClass();
+            i++;
+        }
+        Method method = obj.getClass().getMethod(methodName, argClasses);
+        return method.invoke(obj, args);
+    }
+
+    private void launchBrowser() {
+        new Thread(
+                        new Runnable() {
+
+                            @Override
+                            public void run() {
+                                try {
+                                    if (!View.isInitialised()) {
+                                        // If we dont set this the HUD wont be enabled, which is a
+                                        // bit pointless
+                                        getExtHUD().setHudCmdlineOptionUsed(true);
+                                        getExtHUD().getHudParam().getConfig().save();
+                                    }
+                                    // Object wd =
+                                    callMethodByReflection(
+                                            getExtSelenium(), "getProxiedBrowserByName", "Firefox");
+
+                                    // callMethodByReflection(wd, "get", "https://zaproxy.org");
+                                } catch (Exception e1) {
+                                    if (!View.isInitialised()) {
```

Comment:
Log always? Should ZAP exit if not able to start the browser when in daemon?

Reply:
I think exiting is a bit too harsh in this case